rust/connlib/clients/android/src/lib.rs

```diff
@@ -90,11 +91,13 @@ fn init_logging(log_dir: PathBuf) {
             .with_tag("connlib"),
     );
 
-    let (file_layer, _guard) = file_logger::layer(log_dir);
+    let (file_layer, guard) = file_logger::layer(log_dir);
 
     // Calling init twice causes a panic; instead use try_init which will fail
     // gracefully if this is called more than once.
     let _ = tracing_subscriber::registry().with(file_layer).try_init();
+
```

Comment:
So It hink there might be a problem here, since this can fail the new guard might not correspond to the actually installed global default subscriber. And the old guard was dropped whenever `Session` was droppped(due to a disconnect or something similar) so after that happens, whenever the new guard is dropped events won't be flushed into the file.

I don't think this is a critical problem if the file layer can keep recording events even after the guard is dropped.

Reply:
We should be able to apply the same treatment to the global logger to using https://docs.rs/tracing-subscriber/0.3.17/tracing_subscriber/util/trait.SubscriberInitExt.html#method.set_default.

Could be a worth a try to be consistent.

Reply:
Hmm `set_default` only works for the current scope and not the global scope... I tried using `set_default` over `try_init` and it seems that prevents all logs from being printed. I'll poke around a bit more.

Reply:
That is what I suspected :/

Reply:
I updated this to allow the guard to `Drop` if the `try_init` fails.